### PR TITLE
GOV-389 Subscribe to a eventType logic added

### PR DESCRIPTION
**Pull request by danishjamal104**
Updates in this PR
1. New service which will run on each server startup and create a subscription in SS for receiving updates from requester BB publisher. 
2. Added config for SS.
3. Added config for pub/sub.

[GOV-545](https://mifosforge.jira.com/browse/GOV-545)

[GOV-545]: https://mifosforge.jira.com/browse/GOV-545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by fynmanoj** [comment]
please rebase

File: src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    private String dstOperationId;
+
+    private String delivery;
+
+    private SubscriptionOptionsDTO options;
+
+    public static Builder builder = new Builder();
+
+    public static class Builder {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java:36]: good to see this builder here. However, there are some suggestions
1. naming: use the name of the builder as SubscriptionDTOBuilder.
2. naming: use the word 'with' or 'set' for the setter methods. eg:  withRoomClass(..) or withRoomCode(..) I prefer with over set in Builder
3. Use case: If you are not able to find the need of writing multiple constructors to a class, The builder pattern might be an overhead.
here the SubscriptionDTO is used only in one place in the code ( instantiated only in one way), si IMO, a builder pattern is not needed here, unless you foresee multiple use cases for this class in the future with a lot of member variables being optional.

File: src/main/java/org/mifos/processor/bulk/schema/SubscriptionOptionsDTO.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @JsonProperty("TTL")
+    private int tTL;
+
+    @JsonProperty("FIFO")
+    private boolean fIFO;
+
+    public static Builder builder = new Builder();
+
+    public static class Builder {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/schema/SubscriptionOptionsDTO.java:20]: Same comment as above. 
Additionally:  a class with just two instance variables usually need not call a builder unless these two are optional fields.
In this case, however, I accept the builder pattern as the builder is used for setting the default values for each instance variable.

**Review by fynmanoj — CHANGES_REQUESTED** [review]
Accept or justify on the suggetions

File: src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java (hunk 1 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java:36]: This is part of IM BB story and this DTO might be used at multiple places in future thats why I took this LLD decision of creating a builder pattern.

File: src/main/java/org/mifos/processor/bulk/schema/SubscriptionOptionsDTO.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/schema/SubscriptionOptionsDTO.java:20]: I added builder patter for the default value.

File: src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java (hunk 1 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java]: Updated the builder class name.

File: src/main/java/org/mifos/processor/bulk/schema/SubscriptionOptionsDTO.java (hunk 2 above)
Comment (danishjamal104) [reply src/main/java/org/mifos/processor/bulk/schema/SubscriptionOptionsDTO.java]: updated the builder class name.

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**